### DISCUSSION TITLE
Connect update hook to Change Streams

### DISCUSSION THREAD

**Discussion by Neschastje** [Kamva/mgm#62] [Ideas]
As I see, update hook is called after Update of single document, but doesn't work in case of UpdateMany. It seems like it will be useful to watch changes using Mongo Change Streams, to apply hook even in UpdateMany case and add changed doc as an argument for hook.

**Comment by mehran-prs** [comment]
Hi @Kozyyy 
Thanks for your comment.
I think this is out of the scope of this library. Maybe it's good to let users implement their own watchers' depending on their needs.
As MGM is just a wrapper around the Mongo Go Driver, all collections have the `Watch` method already, so we can just watch on them when we needed.
